server/routes/api/disaster_api.py

```diff
@@ -49,21 +50,134 @@ def event_date_range():
   return Response(json.dumps(result), 200, mimetype='application/json')
 
 
+def keep_event(event, place, filter_prop, filter_unit, filter_upper_limit,
+               filter_lower_limit):
+  """
+  Returns whether or not we should keep an event
+  """
+  if place != EARTH_DCID and not place in event.get("affectedPlaces", []):
+    # event does not affect the place of interest
+    return False
+  if not filter_prop or not filter_prop in event:
+    # no props to filter by or event doesn't contain the prop information
+    return True
+  prop_val = event[filter_prop]
+  if len(prop_val) < len(filter_unit):
+    # value for the property to filter by is empty
+    return True
+  val_string = prop_val[len(filter_unit):].strip()
+  try:
+    val = float(val_string)
+    return val <= filter_upper_limit and val >= filter_lower_limit
+  except:
+    # can't parse the value to filter by into a float
+    return True
+
+
+@bp.route('/json-event-data')
+def json_event_data():
+  """Gets the event data from saved jsons for a given eventType, date, place,
+     andfilter information (filter prop, unit, lower limit, and upper limit).
+
+  Returns: an object of the following form
+    {
+      eventCollection: {
+        events: [
+          {
+            dcid: string,
+            dates: list of strings,
+            geoLocations: list of { point: { latitude: number, longitude: number } },
+            places: list of strings,
+            provenanceId: string,
+            propVals: {
+              [prop]: { vals: list of strings },
+              ...
+            }
+          },
+          ...
+        ],
+        provenanceInfo: {
+          [provenanceId]: {
+            domain: string,
+            importName: string,
+            provenanceUrl: string
+          },
+          ...
+        }
+      }
+  """
+  event_type = request.args.get('eventType', '')
+  if not event_type:
+    return "error: must provide a eventType field", 400
+  date = request.args.get('date', '')
+  if not date:
+    return "error: must provide a date field", 400
+  place = request.args.get('place', '')
+  if not place:
+    return "error: must provide a place field", 400
+  filter_prop = request.args.get('filterProp', '')
+  filter_unit = request.args.get('filterUnit', '')
+  filter_upper_limit = float(request.args.get('filterUpperLimit', '0'))
+  filter_lower_limit = float(request.args.get('filterLowerLimit', '0'))
```

Comment:
should we default to sys.float_info.max or  float("inf") and -float("inf")

Reply:
we can, but defaulting to 0 makes the logic easier & the only time we will have to use default is if we get a filter_prop but no upper and lower limits which shouldn't happen .. I can make the change if you feel strongly